ZenPacks/zenoss/LinuxMonitor/modeler/plugins/zenoss/cmd/linux/lvm.py

```diff
@@ -155,157 +192,64 @@ def process(self, device, results, log):
-        # device (o)pen, (X) unknown
-        return {'o': 'open',
-                'X': 'unknown'}.get(att, None)
-
-    def lv_target_type(self, att):
-        # Target  type: (C)ache, (m)irror, (r)aid, (s)napshot, (t)hin, (u)nknown, (v)irtual
-        return {'C': 'cache',
-                'm': 'mirror',
-                'r': 'raid',
-                's': 'snapshot',
-                't': 'thin',
-                'u': 'unknown',
-                'v': 'virtual'}.get(att, None)
-
-    def lv_health(self, att):
-        # Volume Health: (p)artial, (r)efresh needed, (m)ismatches exist, (w)ritemostly, (X) unknown
-        return {'p': 'partial',
-                'r': 'refresh needed',
-                'm': 'mismatches exist',
-                'w': 'writemostly',
-                'X': 'unknown'}.get(att, None)
\ No newline at end of file
+    def makeHDMap(self, columns):
+        hd_om = ObjectMap()
+        hd_om.title = columns['disk']
+        hd_om.id = 'disk-{}'.format(self.prepId(columns['disk']))
+        hd_om.size = int(columns['size'])
+        hd_om.relname = 'harddisks'
+        hd_om.modname = 'ZenPacks.zenoss.LinuxMonitor.HardDisk'
+        return hd_om
+
+    def makePVMap(self, columns):
+        # pv_name,pv_fmt,pv_attr,pv_size,pv_free,pv_uuid,vg_name
+        pv_om = ObjectMap()
+        pv_om.title = columns['pv_name']
+        pv_om.id = 'pv-{}'.format(self.prepId(columns['pv_name']))
+        pv_om.format = columns['pv_fmt']
+        pv_om.attributes = self.lvm_parser.pv_attributes(columns['pv_attr'])
+        pv_om.pvsize = int(columns['pv_size'])
+        pv_om.free = int(columns['pv_free'])
+        pv_om.uuid = columns['pv_uuid']
+        pv_om.set_volumeGroup = 'vg-{}'.format(columns['vg_name'])
+        pv_om.relname = 'physicalVolumes'
+        pv_om.modname = 'ZenPacks.zenoss.LinuxMonitor.PhysicalVolume'
+        return pv_om
+
+    def makeVGMap(self, columns):
+        # vg_name,vg_attr,vg_size,vg_free,vg_uuid
+        vg_om = ObjectMap()
+        vg_om.title = columns['vg_name']
+        vg_om.id = 'vg-{}'.format(self.prepId(columns['vg_name']))
+        vg_om.attributes = self.lvm_parser.vg_attributes(columns['vg_attr'])
+        vg_om.vgsize = int(columns['vg_size'])
+        vg_om.freesize = int(columns['vg_free'])
+        vg_om.uuid = columns['vg_uuid']
+        vg_om.relname = 'volumeGroups'
+        vg_om.modname = 'ZenPacks.zenoss.LinuxMonitor.VolumeGroup'
+        return vg_om
+
+    def makeLVMap(self, columns):
+        # lv_name,vg_name,lv_attr,lv_size,lv_uuid,origin
+        lv_om = ObjectMap()
+        lv_om.title = columns['lv_name']
+        lv_om.vgname = columns['vg_name']
+        lv_om.id = 'lv-{}'.format(self.prepId(columns['vg_name'])+'_'+self.prepId(columns['lv_name']))
+        lv_om.attributes = self.lvm_parser.lv_attributes(columns['lv_attr'])
+        lv_om.lvsize = int(columns['lv_size'])
+        lv_om.active = True if 'active' in lv_om.attributes else False
```

Comment:
Setting `lv_om.active` is resulting in the following warning in zenhub.log during modeling.

```
2016-02-24 19:00:47,036 INFO zen.ZenHub: Worker (307) reports 2016-02-24 19:00:47,036 WARNING zen.ApplyDataMap: getter for 'active' not found on obj 'lv-centos_home', skipping
```

This is because zenpack.yaml defines LogicalVolume.active as a *datapoint* property. This makes it a method and not something that can be modeled.